docs/site/Interceptors.md

```diff
@@ -0,0 +1,529 @@
+  @intercept('log', logSync)
+  async greetWithTwoInterceptors(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  // No interceptors are attached
+  async greetWithoutInterceptors(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  // Apply `convertName` to convert `name` arg to upper case
+  @intercept(convertName)
+  async greetWithUpperCaseName(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  // Apply `name-validator` backed by a provider class
+  @intercept('name-validator')
+  async greetWithNameValidation(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  // Apply `logError` to catch errors
+  @intercept(logError)
+  async greetWithError(name: string) {
+    throw new Error('error: ' + name);
+  }
+}
+```
+
+### Class level interceptors
+
+To apply interceptors to be invoked for all methods on a class, we can use
+`@intercept` to decorate the class. When a method is invoked, class level
+interceptors (if not explicitly listed at method level) are invoked before
+method level ones.
+
+```ts
+// Apply `log` to all methods on the class
+@intercept(log)
+class MyControllerWithClassLevelInterceptors {
+  // The class level `log` will be applied
+  static async greetStatic(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(log)
+  static async greetStaticWithDI(@inject('name') name: string) {
+    return `Hello, ${name}`;
+  }
+
+  // We can apply `@intercept` multiple times on the same method
+  // This is needed if a custom decorator is created for `@intercept`
+  @intercept(log)
+  @intercept(logSync)
+  greetSync(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  // Apply multiple interceptors. The order of `log` will be preserved as it
+  // explicitly listed at method level
+  @intercept(convertName, log)
+  async greet(name: string) {
+    return `Hello, ${name}`;
+  }
+}
+```
+
+### Global interceptors
+
+Global interceptors are discovered from the `InvocationContext`. They are
+registered as bindings with `interceptor` tag. For example,
+
+```ts
+import {asInterceptor} from '@loopback/context';
+
+app
+  .bind('interceptors.MetricsInterceptor')
+  .toProvider(MetricsInterceptorProvider)
+  .apply(asInterceptor);
+```
+
+### Order of invocation for interceptors
+
+Multiple `@intercept` decorators can be applied to a class or a method. The
+order of invocation is determined by how `@intercept` is specified. The list of
+interceptors is created from top to bottom and from left to right. Duplicate
+entries are removed from their first occurrences.
+
+Let's examine the list of interceptors invoked for each method on
+`MyControllerWithClassLevelInterceptors`, which has a class level `log`
+decorator:
+
+```ts
+// Apply `log` to all methods on the class
+@intercept(log)
+class MyControllerWithClassLevelInterceptors {
+  // ...
+}
+```
+
+1. greetStatic: [`log`]
+
+   ```ts
+   // No explicit `@intercept` at method level. The class level `log` will
+   // be applied
```

Comment:
Please move the content from code comment into regular documentation text.

Same comment applies to other items below too.